### PR TITLE
Update warning logs

### PR DESCRIPTION
**Pull request by molly-moen**
We discussed adding a view into warnings in the javabuilder dashboard in [this PR](https://github.com/code-dot-org/javabuilder/pull/367#discussion_r965243906). When I looked into this I found our warnings are hard to parse because they are generally very long strings/stack traces. This PR makes them easier to group in logs by adding a `type` field and either a `detail` field or exception-specific fields such as `stackTrace`. This PR also moves us towards only using `LoggerUtils` to generate logs.

Tested on a dev instance.

### CODE REVIEW

**Review by madelynkasula — APPROVED** [review]

File: org-code-javabuilder/protocol/src/main/java/org/code/protocol/LoggerUtils.java [hunk 1]
```diff
@@ -101,17 +103,29 @@ public static void logSevereException(Throwable e) {
    */
   public static void logTrackingException(Throwable e) {
```
Comment (sanchitmalhotra126) [inline org-code-javabuilder/protocol/src/main/java/org/code/protocol/LoggerUtils.java:104]: Could we go ahead and change this method name to something warning specific? maybe something like `logExceptionAsWarning` or even just another `logWarning`?

**Review by sanchitmalhotra126 — COMMENTED** [review]
Nice update! Just one comment on a method name

File: org-code-javabuilder/protocol/src/main/java/org/code/protocol/LoggerUtils.java (hunk 1 above)
Comment (molly-moen) [reply org-code-javabuilder/protocol/src/main/java/org/code/protocol/LoggerUtils.java:104]: good idea, will update
